pkg/agent/agent.go

```diff
@@ -397,3 +418,236 @@ func (f *Flows) onInterfaceAdded(iface ifaces.Interface) {
+	graph, err := p.buildAndStartPipeline(ctx)
+	if err != nil {
+		return fmt.Errorf("starting processing graph: %w", err)
+	}
+
+	p.status = StatusStarted
+	plog.Info("Packets agent successfully started")
+	<-ctx.Done()
+
+	p.status = StatusStopping
+	plog.Info("stopping Packets agent")
+	if err := p.ebpf.Close(); err != nil {
+		plog.WithError(err).Warn("eBPF resources not correctly closed")
+	}
+
+	plog.Debug("waiting for all nodes to finish their pending work")
+	<-graph.Done()
+
+	p.status = StatusStopped
+	plog.Info("Packets agent stopped")
+	return nil
+}
+
+func (p *Packets) Status() Status {
+	return p.status
+}
+
+func (p *Packets) interfacesManager(ctx context.Context) error {
+	slog := plog.WithField("function", "interfacesManager")
+
+	slog.Debug("subscribing for network interface events")
+	ifaceEvents, err := p.interfaces.Subscribe(ctx)
+	if err != nil {
+		return fmt.Errorf("instantiating interfaces' informer: %w", err)
+	}
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				slog.Debug("stopping interfaces' listener")
+				return
+			case event := <-ifaceEvents:
+				slog.WithField("event", event).Debug("received event")
+				switch event.Type {
+				case ifaces.EventAdded:
+					p.onInterfaceAdded(event.Interface)
+				case ifaces.EventDeleted:
+					// qdiscs, ingress and egress filters are automatically deleted so we don't need to
+					// specifically detach them from the ebpfFetcher
+				default:
+					slog.WithField("event", event).Warn("unknown event type")
+				}
+			}
+		}
+	}()
+
+	return nil
+}
+
+func (p *Packets) buildAndStartPipeline(ctx context.Context) (*node.Terminal[[]*flow.PacketRecord], error) {
+
+	plog.Debug("registering interfaces' listener in background")
+	err := p.interfacesManager(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	plog.Debug("connecting packets' processing graph")
+
+	perfTracer := node.AsStart(p.perfTracer.TraceLoop(ctx))
+
+	ebl := p.cfg.ExporterBufferLength
+	if ebl == 0 {
+		ebl = p.cfg.BuffersLength
+	}
+
+	packetbuffer := node.AsMiddle(p.packetbuffer.PBuffer,
+		node.ChannelBufferLen(p.cfg.BuffersLength))
+
+	perfTracer.SendsTo(packetbuffer)
+
+	export := node.AsTerminal(p.exporter,
+		node.ChannelBufferLen(ebl))
+
+	packetbuffer.SendsTo(export)
+
+	plog.Debug("starting graph for packet stream")
+	perfTracer.Start()
+
+	return export, nil
+}
+
+func (p *Packets) onInterfaceAdded(iface ifaces.Interface) {
+	// ignore interfaces that do not match the user configuration acceptance/exclusion lists
+	if !p.filter.Allowed(iface.Name) {
+		plog.WithField("interface", iface).
+			Debug("[PANO]interface does not match the allow/exclusion filters. Ignoring")
+		return
+	}
+	plog.WithField("[PANO]interface", iface).Info("interface detected. Registering packets ebpfFetcher")
+	if err := p.ebpf.Register(iface); err != nil {
+		plog.WithField("[PANO]interface", iface).WithError(err).
+			Warn("can't register packet ebpfFetcher. Ignoring")
+		return
+	}
+}
```

Comment:
Overall I feel like there is high level of duplication between flows agent and pano agent so it makes sense to remove it and instantiate through a common agents interface. Else we will have big overhead of code maintenance.

Reply:
@shach33 is this comment to be fixed later? or was it missed.